docs/how-to/add-the-nginx-route-relation.md

```diff
@@ -0,0 +1,146 @@
+charmcraft init
+```
+You'll then see the following output:
+```
+Charmed operator package file and directory tree initialised.
+
+Now edit the following package files to provide fundamental charm metadata and other information:
+
+metadata.yaml
+config.yaml
+src/charm.py
+README.md
+```
+Let's start by updating `metadata.yaml`. We want to update the `display-name` to "My Charm", and the `summary` and `description` to "A charm with a relation to nginx ingress integrator".
+
+We can leave the other files as is for now, but if we want to publish this charm later, we'll want to update them before doing so.
+
+## Deploy the charm
+
+Now let's deploy the charm just to confirm everything is working as expected without the relation. To do that, just run this:
+```
+charmcraft pack
+```
+This will build the charm inside an LXC for you. The output will tell you the location of the built charm. In my case, that was `my-charm_ubuntu-22.04-amd64.charm`.
+
+Now let's add a juju model and deploy our charm.
+```
+juju add-model ingress-test
+juju deploy ./my-charm_ubuntu-22.04-amd64.charm --resource httpbin-image=kennethreitz/httpbin
+```
+To inspect the deployment, let's run `juju status`. Once the application reaches a status of `active idle` our application has been deployed. We can visit it in a browser by getting the IP address of the unit and then going to `http://${ip_of_unit}`.
+
+Note that `juju status` includes two IP addresses, one for the "Unit" and one for the "App". Here's an example:
+```
+Model         Controller          Cloud/Region        Version  SLA          Timestamp
+ingress-test  microk8s-localhost  microk8s/localhost  2.9.37   unsupported  12:12:17+01:00
+
+App       Version  Status  Scale  Charm     Channel  Rev  Address         Exposed  Message
+my-charm           active      1  my-charm             0  10.152.183.205  no       
+
+Unit         Workload  Agent  Address       Ports  Message
+my-charm/0*  active    idle   10.1.129.139         
+```
+
+So we now have a working charm, great! However, what we don't currently have is ingress for our application configured. MicroK8s sets up networking in such a way that you can reach the IPs of units directly, but in a production Kubernetes cluster things don't work in this way. Also, we're visiting a cluster-internal IP address directly, what if we want a real hostname/IP address for this? We'll need to configure Ingress for this to be possible.
+
+Also, you may notice that you can visit the Unit IP address in a browser, but not the App IP address. Why is that? The reason is that the App IP address refers to a [Kubernetes Service](https://kubernetes.io/docs/concepts/services-networking/service/) so as you add other units to the charm they would in theory be reachable through the same IP. However, Juju doesn't yet have a mechanism for a charm to define what port that Service should be configured with, and this why we can't use it to browse the web site (which is listening on port 80).
```

Comment:
by saying juju doesn't have this feature yet, are we implying that it is coming in the future?